### PR TITLE
updated gitignore to include yarn, vscode, webstorm, react

### PR DESCRIPTION
**Pull request by Caopi**
extended gitignore by autogenerated one from gitignore.io for react,webstorm,vscode. also added yarn gitignore settings from https://yarnpkg.com/advanced/qa#which-files-should-be-gitignored. Committed the yarn.lock file which should be part of the repository as described on the same page

### CODE REVIEW

**Comment by Caopi** [comment]
Should fix https://github.com/kenodressel/quarantine-hero/issues/73

File: .gitignore [hunk 1]
```diff
@@ -1,15 +1,17 @@
...
 # testing
-/coverage
+coverage
 
 # production
-/build
+build
 
 # misc
 .DS_Store
```
Comment (kenodressel) [inline .gitignore]: covered by `.DS_*`

File: .gitignore [hunk 2]
```diff
@@ -23,4 +25,126 @@ yarn-debug.log*
...
+.idea/**/sonarlint/
+
+# SonarQube Plugin
+.idea/**/sonarIssues.xml
+
+# Markdown Navigator plugin
+.idea/**/markdown-navigator.xml
+.idea/**/markdown-navigator/
+
+# End of https://www.gitignore.io/api/react,webstorm,visualstudiocode
```
Comment (kenodressel) [inline .gitignore]: should also include `dist`

File: .gitignore [hunk 3]
```diff
@@ -23,4 +25,126 @@ yarn-debug.log*
...
+### VisualStudioCode Patch ###
+# Ignore all local history of files
+.history
+
+### WebStorm ###
+# Covers JetBrains IDEs: IntelliJ, RubyMine, PhpStorm, AppCode, PyCharm, CLion, Android Studio and WebStorm
+# Reference: https://intellij-support.jetbrains.com/hc/en-us/articles/206544839
+
+# User-specific stuff
+.idea/**/workspace.xml
```
Comment (kenodressel) [inline .gitignore]: instead of only blacklisting certain IDE files we should use `.idea`

File: .gitignore (hunk 1 above)
Comment (Caopi) [reply .gitignore]: fixed

File: .gitignore (hunk 2 above)
Comment (Caopi) [reply .gitignore]: fixed

File: .gitignore (hunk 3 above)
Comment (Caopi) [reply .gitignore]: ok

**Pull request merged**